Packages/Core/Runtime/Conditions/AtomCondition.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using UnityEngine;
+
+namespace UnityAtoms
+{
+    /// <summary>
+    /// Base abstract class for Conditions. Inherits from `AtomFunction` and 
+    /// must implement `Condition` Predicate
+    /// </summary>
+    public abstract class AtomCondition<T> : AtomFunction<T, T>
+    {
+        public Predicate<T> Predicate;
+
+        public AtomCondition() => Predicate = Condition;
+        public abstract bool Condition(T value);
```

Comment:
what is the purpose of Condition? is it used? have not seen the usage of it yet.

Reply:
See the first picture in the PR description, it is the actual condition evaluation method to implement which gets assigned to the Predicate. Each extended <Type>Condition needs their own implementation of Condition(T value) to work. I went this way because Predicate <T>is a delegate. If it looks, odd, it's because I've never used it, maybe I got it wrong.

Reply:
AtomFunction<T> defines a method `virtual T Call`, because AtomCondition<T> should inherit from AtomFunction<bool, T> the implementation should override the method `bool Call(T t1)`

you can see this in: https://github.com/AdamRamberg/unity-atoms/blob/master/Packages/BaseAtoms/Runtime/Functions/Clamp/ClampInt.cs

Reply:
Ah, now I get it, thanks!